scripts/clippy.args

```diff
@@ -14,6 +14,9 @@ allowed_lints=(
     "-A" "clippy::needless_lifetimes"
     # This ends up being platform dependent, e.g. the Instant type
     "-A" "clippy::trivially-copy-pass-by-ref"
+
+    # Added when bumping compiler to 1.39 beta track
```

Comment:
I'm not sure in the future when people are wondering why this is here that when it was added will be the main concern. Perhaps we could add why this is added?

Reply:
Yeah my only intent was to sort of explain when it was added, I don't want to also go through the whole code base and fix this new lint that is introduced at the same time as doing a bump.